haystack/document_stores/opensearch.py

```diff
@@ -171,15 +171,15 @@ def __init__(
         if knn_engine not in {"nmslib", "faiss"}:
             raise ValueError(f"knn_engine must be either 'nmslib' or 'faiss' but was {knn_engine}")
 
-        if knn_engine == "faiss" and similarity not in {"dot_product", "l2"}:
-            raise ValueError(
-                f"knn_engine=`faiss` was set to similarity {similarity}. Currently, we only support 'dot_product' and 'l2' similarities. Set the similarity to one of the supported values."
-            )
-
         self.knn_engine = knn_engine
         self.embeddings_field_supports_similarity = False
-        self.similarity_to_space_type = {"cosine": "cosinesimil", "dot_product": "innerproduct", "l2": "l2"}
-        self.space_type_to_similarity = {v: k for k, v in self.similarity_to_space_type.items()}
+        self.similarity_to_space_type = {
```

Comment:
Consider renaming to `similarity_space_type_mappings` and `space_type_similarity_mappings`.